fpmsyncd/routesync.cpp

```diff
@@ -581,7 +588,8 @@ void RouteSync::onMsg(int nlmsg_type, struct nl_object *obj)
     char master_name[IFNAMSIZ] = {0};
 
     /* if the table_id is not set in the route obj then route is for default vrf. */
-    if (master_index)
+    if (master_index &&
+        (master_index != RT_TABLE_MAIN) && (master_index != RT_TABLE_DEFAULT))
```

Comment:
what trigger add of the new check ` (master_index != RT_TABLE_MAIN) && (master_index != RT_TABLE_DEFAULT)`

Reply:
any comment on this ?

Reply:
@abdosi You were inadvertently left off invitation list for live code review where all this was explained.  I do not have you email address to forward you the latest email thread.

Reply:
@abdosi:  I have added a new runtime option "fpmsyncd -l net" that allows fpmsyncd to listen to Netlink messages from Linux kernel directly instead of listening to FRR FPM TCP socket.  This is useful because FRR does not send any MPLS info (not even encap MPLS NH info for IP routes) on its FPM TCP socket.  While both Juniper and Metaswitch have alternate routing stacks to test without FRR, the community in general does not have a working open-source routing-stack to verify MPLS functionality.  This "-l net" option is the way I was able to unit-test the full SONiC MPLS code path independent of routing-stack implementation and I have used this for the unit-tests I have added to sonic-swss/tests.  The main difference between Netlink messages from FRR and from Linux kernel is that SONiC has modified FRR FPM Netlink to substitute VRF ifindex for RTA_ROUTE_TABLE, whereas Linux kernel is unmodified in SONiC and will send route table ID.  This check that you are asking about will make sure that Netlink messages from kernel are not incorrectly treated as VRF ifindex.

Reply:
thanks for detail explanation

Reply:
This content has been moved to a new sonic-swss PR (https://github.com/Azure/sonic-swss/pull/1765)